### ISSUE TITLE
When Tor client is open and initialized,  is it only using specific ports to set up the circuit? like 9001 9004 and others?  

### ISSUE THREAD

**Issue by GitZinger** [shadow/shadow#768]
**State your question concisely**
When Tor client is open and initialized,  is it only using specific ports to set up the circuit? like 9001 9004 and others?  
according to https://2019.www.torproject.org/docs/faq.html.en#FileSharing 
I mean When Tor client initializes the circuit with the entry of relays, only 9001 9004 and 4443 ports are used? or it could use random ports totally anytime?

**Describe your current understanding of Shadow as it relates to your question**

if the Tor client cannot initialize the circuit blocked by a firewall or something, then it won't work right?

**Describe which parts you are unsure about**
Tor client needs to first initialize a circuit before users browse the webpages right? and this initialization is up to the setup of the circuit?

**Comment by robgjansen** [comment]
It seems you are asking about how client port allocation works. Servers are available at well known ports, but clients generally get assigned a random port by the operating system.
https://en.wikipedia.org/wiki/Ephemeral_port

Also, please don't spam the Shadow developers with general questions about Tor or Linux networking - you can search the Internet for those.

**Comment by GitZinger** [comment]
Sorry. did intent to spam. I am thinking to use the ports to block the traffic or communication between Tors in the shadow. 
Then I encounter two problems. One is that there is no specific route and connection between nodes and hosts. for which you just answered me in another issue.
Another one is that in socket programming, I might try to block the Tor communication based on the ports. That is why I posted this question. Sorry for the inconvenience.
